src/balanceof_imbalances.py

```diff
@@ -66,35 +64,43 @@ def get_balances(
         self, token_addresses: Set[ChecksumAddress], block_number: int
     ) -> Dict[ChecksumAddress, Optional[int]]:
         """Get balances for all tokens at the given block number."""
-        balances = {}
-        balances[NATIVE_ETH_TOKEN_ADDRESS] = self.get_eth_balance(
-            SETTLEMENT_CONTRACT_ADDRESS, block_number
+        balances: Dict[ChecksumAddress, Optional[int]] = {}
+        balances[
+            self.web3.to_checksum_address(NATIVE_ETH_TOKEN_ADDRESS)
+        ] = self.get_eth_balance(
+            self.web3.to_checksum_address(SETTLEMENT_CONTRACT_ADDRESS), block_number
         )
 
         for token_address in token_addresses:
             balances[token_address] = self.get_token_balance(
-                token_address, SETTLEMENT_CONTRACT_ADDRESS, block_number
+                token_address,
+                self.web3.to_checksum_address(SETTLEMENT_CONTRACT_ADDRESS),
+                block_number,
             )
 
         return balances
 
     def calculate_imbalances(
         self,
-        prev_balances: Dict[str, Optional[int]],
-        final_balances: Dict[str, Optional[int]],
-    ) -> Dict[str, int]:
+        prev_balances: Dict[ChecksumAddress, Optional[int]],
+        final_balances: Dict[ChecksumAddress, Optional[int]],
+    ) -> Dict[ChecksumAddress, int]:
         """Calculate imbalances between previous and final balances."""
-        imbalances = {}
+        imbalances: Dict[ChecksumAddress, int] = {}
         for token_address in prev_balances:
             if (
                 prev_balances[token_address] is not None
                 and final_balances[token_address] is not None
             ):
-                imbalance = final_balances[token_address] - prev_balances[token_address]
+                prev_balance = prev_balances[token_address]
+                assert prev_balance is not None  # Ensure prev_balance is not None
```

Comment:
I think it might be better to comment why we need to make this check. Eg. 'prev_balance cannot be None because ...' or 'prev_balance may be None because'.